model.py

```diff
@@ -339,7 +345,7 @@ def search_by_location_name(cls, _db, query, type=None, here=None):
         # For a library to match, the Place named by the query must
         # intersect a Place served by that library.
         named_place = aliased(Place)
-        qu = _db.query(Library).join(
+        qu = _db.query(Library).distinct().join(
```

Comment:
Without this distinct() call, a library will show up more than once it it serves two different areas, and those areas are different distances away from `here`. This wasn't a problem when we weren't selecting `distance`.